doc/whats_new/v1.4.rst

```diff
@@ -316,6 +316,16 @@ Changelog
   dtype as categories in the algorithm. `categorical_features="from_dtype"` will become
   the default in v1.6. :pr:`26411` by `Thomas Fan`_.
 
+- |Feature| the `categorical_features="from_dtype"` option of
+  :class:`ensemble.HistGradientBoostingClassifier` and
+  :class:`ensemble.HistGradientBoostingRegressor` described above is also
+  supported for other dataframes supporting the dataframe interchange protocol,
+  such as polars dataframes. Categorical features no longer need to be encoded
```

Comment:
Unfortunately it does not work with pyarrow as it is, because the HistGradientBoosting relies on the ColumnTransformer which does not seem to support pyarrow:

```python
import pyarrow as pa
from sklearn.compose import ColumnTransformer

x = pa.Table.from_pydict({"a": [1.0, 1.0]})
ct = ColumnTransformer([("a", "passthrough", ["a"])])
ct.fit_transform(x)  # TypeError
```

Reply:
Ok let's do this in a follow-up PR to `ColumnTransformer` then.